### PR TITLE
Big-endian fix: Accessing OBJECTHEADER in ImageConverter

### PR DESCRIPTION
**Pull request by uweigand**
The GetBitmapStream routine in ImageConverter is supposed to
skip an embedded OBJECTHEADER structure.  However, this
structure needs to be read in little-endian mode, not in
native byte order.

Fixes the ImageConverterTest.ImageWithOleHeader test on
big-endian systems.

### CODE REVIEW

**Review by safern — APPROVED** [review]
LGTM, thanks!

File: src/libraries/System.Drawing.Common/src/System/Drawing/ImageConverter.cs [hunk 1]
```diff
@@ -112,20 +113,21 @@ public override PropertyDescriptorCollection GetProperties(ITypeDescriptorContex
...
 
                 if (signature != 0x1c15)
                 {
                     return null;
                 }
 
                 // The data is in the form of OBJECTHEADER. It's an encoded format that Access uses to push imagesinto the DB.
-                OBJECTHEADER pHeader = MemoryMarshal.Read<OBJECTHEADER>(rawData);
+                // We only use the headersize, but it needs to be read as little-endian data.
+                short headersize = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(2, 2));
```
[the remark this answers is not in the document]
Comment (safern) [reply src/libraries/System.Drawing.Common/src/System/Drawing/ImageConverter.cs:125]: I would say let's remove it. I did a GH search and didn't find any other usages.

[the remark this answers is not in the document]
Comment (uweigand) [reply src/libraries/System.Drawing.Common/src/System/Drawing/ImageConverter.cs:142]: OK, branch updated accordingly.

File: src/libraries/System.Drawing.Common/src/System/Drawing/ImageConverter.cs [hunk 2]
```diff
@@ -112,20 +113,38 @@ public override PropertyDescriptorCollection GetProperties(ITypeDescriptorContex
...
             {
-                short signature = MemoryMarshal.Read<short>(rawData);
+                short signature = BinaryPrimitives.ReadInt16LittleEndian(rawData);
 
                 if (signature != 0x1c15)
                 {
                     return null;
                 }
 
                 // The data is in the form of OBJECTHEADER. It's an encoded format that Access uses to push imagesinto the DB.
```
Comment (stephentoub) [inline src/libraries/System.Drawing.Common/src/System/Drawing/ImageConverter.cs:123]: Nit (pre-existing):
```suggestion
                // The data is in the form of OBJECTHEADER. It's an encoded format that Access uses to push images into the DB.
```

**Review by stephentoub — APPROVED** [review]
Thanks

**Pull request merged**